CRM/MembershipExtras/Service/MembershipEndDateCalculator.php

```diff
@@ -9,15 +12,68 @@ class CRM_MembershipExtras_Service_MembershipEndDateCalculator {
    * @param int $membershipId
    *
    * @return string
+   * @throws \CiviCRM_API3_Exception
+   * @throws \CiviCRM_API3_Exception|\Exception
    */
   public static function calculate($membershipId) {
-    $membershipDetails = civicrm_api3('Membership', 'get', [
+    $newEndDate = 'null';
+    $membershipDetails = self::getMembership($membershipId);
+    $interval = self::getMembershipPeriodInterval($membershipDetails);
+    if (!empty($interval)) {
+      $currentEndDate = new DateTime($membershipDetails['end_date']);
+      $currentEndDate->add(new DateInterval($interval));
+      $newEndDate = $currentEndDate->format('Ymd');
+    }
+
+    return $newEndDate;
+  }
+
+  /**
+   * Calculates the previous end date for the given membership.
+   *
+   * @param int $membershipID
+   *
+   * @return string
+   * @throws \CiviCRM_API3_Exception|\Exception
+   */
+  public static function calculatePreviousEndDate($membershipID) {
+    $previousEndDate = 'null';
+    $membershipDetails = self::getMembership($membershipID);
+    $interval = self::getMembershipPeriodInterval($membershipDetails);
+
```

Comment:
I don't think you need a new line here, should be consistent with calculate() method.